src/mlpack/methods/matrix_completion/matrix_completion.cpp

```diff
@@ -59,13 +60,7 @@ void MatrixCompletion::CheckValues()
         << "indices does not have 2 rows!" << std::endl;
   }
 
-  if (indices.n_cols != values.n_elem)
-  {
-    Log::Fatal << "MatrixCompletion::CheckValues(): the number of constraint "
-        << "indices (columns of constraint indices matrix) does not match the "
-        << "number of constraint values (length of constraint value vector)!"
-        << std::endl;
-  }
+  util::CheckSameSizes(indices, values, "MatrixCompletion::CheckValues()", "indices");
```

Comment:
Test cases were failing in this part. 
I have finally figured out why. 
Actually the implementation of `util::CheckSameSizes()` was changed a bit in #3047 . 
To be specific - 
`if (data.n_cols != label.n_elem)`  -->  `if (data.n_cols != label.n_cols)`

Now in Matrix Completion from [here](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/matrix_completion/matrix_completion.hpp#L62)
matrix `values` is `p x 1`
matrix  `indices` is `2 x p`

Therefore we are getting error 
`FAILED:
due to unexpected exception with message:
  MatrixCompletion::CheckValues(): number of points (240) does not match number
  of values (1)`


Hotfix : Transpose `value` matrix - ( Considering we will face this situation at very few places ) 
`arma::mat transposeValues = values.t();` 
`util::CheckSameSizes(indices, transposeValues, "MatrixCompletion::CheckValues()", "values");`

@jonpsy what do you think about this ?

Reply:
I think the expectation here is the `values` would be a row vector`(1 x n_rows)`, which is why `labels.n_elem` was changed to `label.n_cols` this is fine. Any challenges to this assumption must be caught ASAP.

Put an assert that `labels.isrowvec()` and throw error appropriately.

Reply:
I think there is some misunderstanding. 

I agree that the expectation of current `util::CheckSameSizes()` is that `values` should be a row vector(1 x n_rows) which is the case in most of use-cases.

But here in matrix completion algorithm structure we are required to check the equality of the  `n_rows` of  `values` (which is a column vector) with `n_cols` of `indices` ( defined as row vector).  

Therefore in order to accommodate this, we have two options which i discussed :
-  Transpose `values` so that it becomes a row vector and then we pass it through `util::SameSizeChecks`
-  Make a new check functionality similar to `util::SameSizeChecks` with difference being 
`if (data.n_cols != label.n_rows)`

Reply:
>     * Transpose `values` so that it becomes a row vector and then we pass it through `util::SameSizeChecks`
>       ( if we assume that such kind of checks would arise in very rare cases)

Of course, perform the transpose. 

What I meant was, since this algorithm expects row vector make sure that assumption is validated at earlier to make for an graceful exit.